src/status_im/contexts/onboarding/events.cljs

```diff
@@ -1,31 +1,192 @@
+         (conj [:dispatch [:push-notifications/switch true]])
+
+         :always
+         (conj [:dispatch
+                [:onboarding/notifications-setup-done
+                 {:onboarding? onboarding?
+                  :syncing?    syncing?
+                  :biometrics? biometrics?}]]))})
+
+(rf/reg-event-fx :onboarding/notifications-setup-start notifications-setup-start)
+
+(defn notifications-setup-done
+  [{:keys [db]} [{:keys [biometrics? onboarding? syncing?]}]]
+  {:fx (cond-> []
+         (and biometrics? onboarding? syncing?)
+         (conj [:dispatch [:onboarding/finalize-setup]])
+
+         (and biometrics? onboarding? (not syncing?))
+         (conj [:dispatch [:onboarding/create-account-and-login]])
+
+         (and (not biometrics?) onboarding? syncing?)
+         (conj [:dispatch [:onboarding/finish-onboarding]])
+
+         (and (not biometrics?) onboarding? (not syncing?))
+         (conj [:dispatch [:onboarding/create-account-and-login]])
+
+         (and (not biometrics?) (not onboarding?) (not syncing?))
+         (conj [:dispatch [:shell/show-root-view]]))})
+
+
+(rf/reg-event-fx :onboarding/notifications-setup-done notifications-setup-done)
+
+(defn notifications-setup
+  [{:keys [db]}
+   [{:keys [biometrics-supported? biometrics? syncing? onboarding?]}]]
+  (let [db-view-id      (get-in db [:view-id])
+        current-view-id (if (= db-view-id :screen/onboarding.syncing-biometric)
+                          :screen/onboarding.enable-biometrics
+                          db-view-id)]
+    {:db (assoc-in db [:onboarding/profile :notifications-prompted?] true)
+     :fx [[:dispatch
+           [:navigate-to-within-stack
+            [:screen/onboarding.enable-notifications current-view-id]
+            {:syncing?    syncing?
+             :onboarding? onboarding?
+             :biometrics? (and biometrics-supported? biometrics?)}]]]}))
+
+(rf/reg-event-fx :onboarding/notifications-setup notifications-setup)
+
+(defn biometrics-setup-start
+  [_ [{:keys [enable-biometrics? syncing?]}]]
+  {:fx (cond-> []
+         enable-biometrics?
+         (conj [:dispatch
+                [:onboarding/enable-biometrics
+                 {:on-done [:onboarding/biometrics-setup-done
+                            {:on-success [:onboarding/notifications-setup
+                                          {:biometrics?           true
+                                           :biometrics-supported? true
+                                           :onboarding?           true
+                                           :syncing?              syncing?}]
+                             :on-fail    [:onboarding/biometrics-fail]}]}]])
+
+         (not enable-biometrics?)
+         (conj [:dispatch
+                [:onboarding/notifications-setup
+                 {:biometrics?           false
+                  :biometrics-supported? true
+                  :onboarding?           true
+                  :syncing?              syncing?}]]))})
```

Comment:
Here's a neat example of how we can compose event vectors together to thread context and state without using callback functions.